stages/org.osbuild.grub.iso

```diff
@@ -0,0 +1,163 @@
+#!/usr/bin/python3
+"""
+Create a boot filesystem tree, can be consumed to create
+an efiboot.img.
+
+"""
+import os
+import re
+import shutil
+import sys
+import tempfile
+import osbuild.api
+
+SCHEMA_2 = """
+"options": {
+  "additionalProperties": false,
+  "required": ["product", "kernel", "isolabel"],
+  "properties": {
+    "product": {
+      "type": "object",
+      "additionalProperties": false,
+      "required": ["name", "version"],
+      "properties": {
+        "name": {"type": "string"},
+        "version": {"type": "string"}
+       }
+    },
+    "kernel": {
+      "type": "string"
+    },
+    "isolabel": {
+      "type": "string"
+    },
+    "efi": {
+      "type": "object",
+      "additionalProperties": false,
+      "required": ["architectures", "vendor"],
+      "properties": {
+        "architectures": {
+            "type": "array",
+            "items": {
+                "type": "string"
+            }
+        },
+        "vendor": {
+            "type": "string"
+        }
+      }
+    },
+    "kernel_opts": {
+      "description": "Additional kernel boot options",
+      "type": "string"
+    },
+    "templates": {
+      "type": "string",
+      "default": "99-generic"
```

Comment:
Only passing in the template subdir locks users into only using the templates under `/usr/share/lorax/templates.d` and it can sometimes (especially when testing new things) be useful to point to arbitrary directories.

Also, whatever constructs this config json will need to search `/usr/share/lorax/templates.d` for the right template to use. The right way to do that is to do it the same way Lorax does and pick the first directory in sorted order (eg. 80-rhel overrides 99-generic).

So I think it's better to just make this a free-form path and let the caller decide the full path to pass in.

Reply:
fixed